extensions/iceberg/src/main/java/io/deephaven/iceberg/internal/Explore.java

```
@@ -242,7 +245,23 @@ public static Table properties(Map<String, String> properties) {
                 .view(properties.entrySet());
     }
 
+    public static Table partitionStats(org.apache.iceberg.Table table) {
+        return partitionStats(table, table.currentSnapshot());
+    }
+
     public static Table partitionStats(org.apache.iceberg.Table table, Snapshot snapshot) {
+        final Collection<PartitionStats> partitionStats;
+        if (snapshot == null) {
+            partitionStats = Collections.emptyList();
```

Nitpick: Should we just throw an exception here? Or make it a @NotNull?

---

No, a newly created table that doesn't have any snapshots is a valid state IMO.

---

I was thinking about the case if a user calls it with null for snapshot, the just get back an empty table.
It feels weird. We can add some javadoc to explain it